lib/pbench/server/cache_manager.py

```diff
@@ -884,14 +926,22 @@ def get_results(self, lock: LockManager) -> Path:
         direct to the caller.
 
         Args:
-            lock:   A lock context manager in shared lock state
+            lock: A lock context manager in shared lock state
 
         Returns:
             the root Path of the unpacked directory tree
         """
 
         if not self.unpacked:
             lock.upgrade()
+
+            # If necessary, attempt to reclaim some unused cache so we have
+            # enough room.
+            if self.controller and self.controller.cache_manager:
+                self.controller.cache_manager.reclaim_cache(
+                    goal_bytes=self.get_unpacked_size()
+                )
```

Comment:
Yeah, but what padding, or rounding factor, is actually guaranteed to be worth the effort? Is it fixed, or should it scale with the total size? In the absence of data (or, I suppose, the motivation to try to generate that data), I didn't see much point.

Reply:
Oh, I don't think there are any guarantees, but my instinct is that it doesn't really need to scale.

We just need to ensure that the "round off errors" (of which there might be several, and I don't know whether they compound or cancel each other...) don't result in us failing to reclaim enough cache such that the subsequent unpack fails when it otherwise wouldn't or shouldn't have.

Rounding it up to the next disk block in size seems like a really good idea, since that's how the allocation will _actually_ be made (although, presumably, the _deallocations_ will also be made in that increment, so once `shutil.disk_usage()` says that there is enough room, there presumably will be).  The problem is that not all disk devices have the same block size, and asking what it is seems like more trouble than it's worth...so, I would just round up to the next Mb or so.

In any case, in the absence of data, I would err on the side of caution:  reclaiming too much cache is a much better result than failing to unpack.